gamemodes/terrortown/gamemode/server/sv_player_ext.lua

```diff
@@ -1155,57 +1178,62 @@ function plymeta:PickupWeapon(wep, dropBlockingWeapon, shouldAutoSelect)
 		end
 	end
 
-	-- this flag is set to the player to make sure he only picks up this weapon
-	self.wpickup_weapon = wep
+	-- prevent race conditions
+	self.wpickup_waitequip[kind] = true
 
-	-- the flag is set to the weapon to stop other players from auto-picking up this weapon
-	wep.wpickup_player = self
+	timer.Create(kind .. "_WeaponPickup_" .. self:SteamID64(), 0, 1, function()
```

Comment:
functions like this should be localized too to save performance

Reply:
Localized as in create a local function outside of PickupWeapon or as in create a local function inside of PickupWeapon (similar to SetWeaponPos)? I would assume the first thing, since in my understanding the later wouldn't improve performance. If so, I should probably also move SetWeaponPos outside of WeaponEquip

Reply:
Both is - regarding to the lua docu - a better performance than doing function declarations just inline. Don't ask me why, I didn't gave a deep look into luajit or stuff like this. Maybe because it isn't optimized and need to do several other actions localizing the inlined function. Maybe not. But answering your question, defining a - maybe abstract / general - function outside this function (as a first instance local function) would lead to the best performance.

Reply:
additionally, there need to be a valid check for self (or the localized thing of self :) )